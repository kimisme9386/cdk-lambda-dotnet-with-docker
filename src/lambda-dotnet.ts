import * as path from 'path';
import * as lambda from '@aws-cdk/aws-lambda';
import * as cdk from '@aws-cdk/core';

export class LambdaDotnet extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props: cdk.StackProps) {
    super(scope, id, props);

    new lambda.DockerImageFunction(this, 'DotnetLambda', {
      code: lambda.DockerImageCode.fromImageAsset(
        path.join(__dirname, '../DemoFunction/src/DemoFunction'),
      ),
    });
  }
}
