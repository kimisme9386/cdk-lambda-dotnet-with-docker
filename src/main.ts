import { App } from '@aws-cdk/core';
import { LambdaDotnet } from './lambda-dotnet';

// for development, use account/region from cdk cli
const devEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};

const app = new App();

new LambdaDotnet(app, 'LambdaDotnetDemo', { env: devEnv });

app.synth();
