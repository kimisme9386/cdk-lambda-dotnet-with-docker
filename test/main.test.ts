import { SynthUtils } from '@aws-cdk/assert';
import '@aws-cdk/assert/jest';
import { App } from '@aws-cdk/core';
import { LambdaDotnet } from '../src/lambda-dotnet';

test('Snapshot', () => {
  const env = {
    region: process.env.CDK_DEFAULT_REGION,
    account: process.env.CDK_DEFAULT_ACCOUNT,
  };

  const app = new App();
  const stack = new LambdaDotnet(app, 'test', { env });

  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
