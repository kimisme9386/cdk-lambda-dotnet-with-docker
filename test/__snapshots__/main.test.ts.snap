// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
Object {
  "Resources": Object {
    "DotnetLambda87B89B9E": Object {
      "DependsOn": Array [
        "DotnetLambdaServiceRole476EB6D6",
      ],
      "Properties": Object {
        "Code": Object {
          "ImageUri": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Ref": "AWS::AccountId",
                },
                ".dkr.ecr.",
                Object {
                  "Ref": "AWS::Region",
                },
                ".",
                Object {
                  "Ref": "AWS::URLSuffix",
                },
                "/aws-cdk/assets:fa160ccfb38f7de49c31758bdb932230d153ffec2886a02fd6f681072e1c494f",
              ],
            ],
          },
        },
        "PackageType": "Image",
        "Role": Object {
          "Fn::GetAtt": Array [
            "DotnetLambdaServiceRole476EB6D6",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "DotnetLambdaServiceRole476EB6D6": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
